android/src/main/java/io/ably/lib/push/SharedPreferenceStorage.java

```diff
@@ -0,0 +1,52 @@
+package io.ably.lib.push;
+
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import java.lang.reflect.Field;
+
+public class SharedPreferenceStorage implements Storage{
+
+    private final ActivationContext activationContext;
+
+    public SharedPreferenceStorage(ActivationContext activationContext) {
+        this.activationContext = activationContext;
+    }
+
+    @Override
+    public void putString(String key, String value) {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(activationContext.getContext());
```

Comment:
This code is being repeated in every interface method, can we take out somewhere?

Reply:
Maybe it can be initialised just once, in constructor of `SharedPreferenceStorage`, however I'm not sure since it was also recreated every time when it was called in `LocalDevice` (original implementation). Or did you just mean to create method for it?

Reply:
I am coming from `DRY` principle, we should take out repetition somewhere else.